File: doc/codes/csharp/examples/api/compute/pattern/laguerre-gaussian.cs
using System.Numerics;
using AUTD3;
using static AUTD3.Units;

namespace DocSamples.ApiComputePatternLaguerreGaussian;

internal static class Sample
{
    internal static void Run()
    {
        var geometry = new Geometry(new[] { new Autd3(Vector3.Zero) });

        var target = geometry.Center + new Vector3(0.0f, 0.0f, 150.0f);
        var axis = Vector3.UnitZ;
        var wavelength = Pattern.Wavelength(340.0f * m / s);
        var phases = geometry.PhaseBuffer();
        var intensities = geometry.IntensityBuffer();

        // ANCHOR: api
        var option = new LaguerreGaussianOption(p: 1, l: 1, waist: 10.0f * mm);
        Pattern.LaguerreGaussianPhase(geometry, target, axis, option, wavelength, phases);
        Pattern.LaguerreGaussianIntensity(geometry, target, axis, option, wavelength, intensities);
        // ANCHOR_END: api
    }
}

File: doc/codes/rust/examples/api/compute/pattern/laguerre-gaussian.rs
use autd3_rs::geometry::{Autd3, Geometry, Vector3, offset};
use autd3_rs::units::{m, mm, s};
use autd3_rs_pattern::{
    LaguerreGaussianOption, laguerre_gaussian_phase, laguerre_gaussian_intensity, wavelength,
};

fn main() {
    let geometry = Geometry::new(vec![Autd3::default()]);

    let target = geometry.center() + offset(0.0 * mm, 0.0 * mm, 150.0 * mm);
    let axis = Vector3::z_axis();
    let wavelength = wavelength(340.0 * m / s);
    let mut dst = geometry.pattern_buffer();

    // ANCHOR: api
    let option = LaguerreGaussianOption {
        p: 1,
        l: 1,
        waist: 10.0 * mm,
    };
    laguerre_gaussian_phase(&geometry, target, axis, option, wavelength, &mut dst);
    laguerre_gaussian_intensity(&geometry, target, axis, option, wavelength, &mut dst);
    // ANCHOR_END: api
}
